### ISSUE TITLE
Consider add an diagnostic to report call to an virtual method via base from an different method?

### ISSUE THREAD

**Issue by yyjdelete** [dotnet/roslyn#38542]
**Version Used**: 
VS2019 16.3.0preview3.0(3.3.1-beta3-19430-03+fa8e2c9b566e4471a3509fc63d7baca0a2a6d30b)
with `Microsoft.CodeAnalysis.FxCopAnalyzers` 2.9.4

**Steps to Reproduce**:

```cs
    public class Class1
    {
        public virtual void Test1()
        {
        }

        /*public void Test2()
        {
        }*/
    }
    public class Class2 : Class1
    {
        public void Test2()
        {
            base.Test1();
        }
    }
    public class Class3 : Class2
    {
        public override void Test1()
        {
            Console.WriteLine("Do something.");
        }
    }
```

**Expected Behavior**:
It will be better if roslyn can report an diagnostic(Info or Warn) for `base.Test1();`(call virtual method `Class1.Test1()` via `base` from another method `Test2()`), it can easily be an mistake of `base.Test2()` or `this.Test1()`(I used to use `base` when there is no `Test1()` in `Class2`, but it will also make the override in `Class3.Test1()` useless).

**Actual Behavior**:
No diagnostic is reported.

**Comment by jaredpar** [comment]
Thanks for the suggestion. Such a diagnostic would be a breaking change and cause diagnostics to be emitted in a significant amount of existing code. 

Irrespective of the breaking change aspect, I'd be against adding a diagnostic here. Yes this can be a mistake but it's also just as likely to be the intended action of the developer.
